NAS-138438 / 26.04 / Protect rule generation against circular links

mgrimesix opened the pull request:
Add exception handler for circular links.

The rule generation script will get stuck if it attempts to process a circular link.  This results in a corrupted rule file.
This is a more rare condition, but has happened.   The directory `/run/udev/watch` is notorious for containing circular links.

Also fixed indentation issue on the parser argument lines.

mgrimesix committed:
Protect rule generation against circular links

bugclerk commented:
Jira URL: https://ixsystems.atlassian.net/browse/NAS-138438

mgrimesix commented:
time 0:45

themylogin commented on rules/privileged-rules.py:
```diff
@@ -53,8 +53,10 @@ def generate_audit_privilege(target_dir: str, privilege_file: str, prefix: str)
-                    # possibly broken symlink
-                    pass
+                    pass  # possibly broken symlink
+                except OSError as OSe:
```
See Python [PEP 8: Function and Variable Names](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names):

Function names should be [lowercase, with words separated by underscores](https://en.wikipedia.org/wiki/Snake_case) as necessary to improve readability.

Variable names follow the same convention as function names.

Exceptions are commonly named just `e`

mgrimesix committed:
Update to be PEP 8 compatible.

mgrimesix committed:
Change to exception variable name to 'e'.

mgrimesix committed:
Replace '40' with errno.ELOOP.
Add exception handlers with reporting.  This should avoid a badly corrupted rule file.

mgrimesix committed:
Update the error logging.

bugclerk commented:
This PR has been merged and conversations have been locked.
If you would like to discuss more about this issue please use our forums or raise a Jira ticket.
